live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/hx_insight/HxInsightEventPublisher.java

```diff
@@ -76,8 +79,9 @@ public void configure()
             .marshal()
             .json()
             .log("Sending event ${body}")
+            .setHeader(USER_AGENT_DATA, applicationInfoProvider::getUserAgentData)
             .process(authService::setHxIAuthorizationHeaders)
-            .to(integrationProperties.hylandExperience().ingester().endpoint())
+            .toD(integrationProperties.hylandExperience().ingester().endpoint() + "&userAgent=${header.user-agent-data}")
```

Comment:
This value should be passed as query param, and as a header?
Also this should be passed only while ingesting metadata?

Reply:
Unfortunately `Camel` has it's own way of setting/modyfing `User-agent` header and it seems only possible by adding `userAgent=<someValue>` path param. It took me a while to figure this out. 
So generally, I am setting a header with key `user-agent-data` and then using this header to dynamically "create" the path with proper param value. It is a bit dirty, but I couldn't find any other way to deal with that - I can try with Exchange property, WDYT?

Reply:
Exchange property sounds great - less confusing.
We had a talk - thanks for explaining that query param `userAgent` will be removed from path by Camel and vale will be passed as HTTP header 👍

Reply:
Probably this header will required by both HxI endpoints (also by `/presigned-urls`) - reference: https://hyland.atlassian.net/wiki/spaces/HxAI/pages/1312928942/ADR+0003+-+Insight+Ingestion+Integration+API+Contract#Required-Headers

Reply:
Let's wait for response from HxI team - if needed - I'll add this in a separate PR - is that ok?

Reply:
Agreed - OK